packages/core/src/services/chains/FreeProviderQuotaChecker/index.ts

```diff
@@ -0,0 +1,42 @@
+import { env } from '@latitude-data/env'
+
+import {
+  DEFAULT_PROVIDER_MAX_FREE_RUNS,
+  ProviderApiKey,
+  RunErrorCodes,
+  Workspace,
+} from '../../../browser'
+import { Result } from '../../../lib'
+import { incrFreeRuns } from '../../freeRunsManager'
+import { ChainError } from '../ChainErrors'
+
+export class FreeProviderQuotaChecker {
+  private workspace: Workspace
+  private provider: ProviderApiKey
+
+  constructor({
+    workspace,
+    provider,
+  }: {
+    workspace: Workspace
+    provider: ProviderApiKey
+  }) {
+    this.workspace = workspace
+    this.provider = provider
+  }
+
+  async call() {
+    if (this.provider.token !== env.DEFAULT_PROVIDER_API_KEY) {
+      return Result.ok(true)
+    }
+    const value = await incrFreeRuns(this.workspace.id)
+    if (value <= DEFAULT_PROVIDER_MAX_FREE_RUNS) return Result.ok(true)
+
+    return Result.error(
+      new ChainError({
+        code: RunErrorCodes.DefaultProviderExceededQuota,
+        message: 'You have exceeded your maximum number of free runs for today',
+      }),
+    )
+  }
+}
```

Comment:
why is this a class now, this is not ruby you don't need constructors to declare a pure function

Reply:
Again. I can change. It's important?

Reply:
It's a bit confusing imo

Reply:
What's confusing? If I change `call` method to a function would it be less confusing?

Reply:
By popular demand now this is a function